packages/mobile/src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,386 @@
+              {t('dappsScreenBottomSheet.title', { dappName: dappSelected?.name })}
+            </Text>
+            <Text style={{ ...fontStyles.regular, textAlign: 'center', flex: 1 }}>
+              {t('dappsScreenBottomSheet.message')}
+            </Text>
+            <Button
+              style={styles.bottomSheetButton}
+              onPress={onPressNavigationButton}
+              text={t('dappsScreenBottomSheet.button', { dappName: dappSelected?.name })}
+            />
+          </View>
+        </View>
+      </BottomSheet>
+
+      <Dialog
+        title={t('dappsScreenHelpDialog.title')}
+        isVisible={isHelpDialogVisible}
+        secondaryActionText={t('dappsScreenHelpDialog.dismiss')}
+        secondaryActionPress={() => setHelpDialogVisible(false)}
+      >
+        {t('dappsScreenHelpDialog.message')}
+      </Dialog>
+      <>
+        {loading && (
+          <View style={styles.centerContainer}>
+            <ActivityIndicator
+              style={styles.loadingIcon}
+              size="large"
+              color={colors.greenBrand}
+              testID="DAppExplorerScreen/loading"
+            />
+          </View>
+        )}
+        {!loading && error && (
+          <View style={styles.centerContainer}>
+            <Text style={fontStyles.regular}>{t('dappsScreen.errorMessage')}</Text>
+          </View>
+        )}
+        {!loading && !error && result && (
+          <SectionList
+            ListHeaderComponent={<DescriptionView message={t('dappsScreen.message')} />}
+            style={styles.sectionList}
+            sections={parseResultIntoSections(result)}
+            renderItem={({ item: category }) => <Dapp dapp={category} onPressDapp={onPressDapp} />}
+            keyExtractor={(item: Dapp) => `${item.categoryId}-${item.id}`}
+            stickySectionHeadersEnabled={false}
+            renderSectionHeader={({ section }: { section: SectionListData<any, SectionData> }) => (
+              <CategoryHeader category={section.category} />
+            )}
+          />
+        )}
+      </>
+    </SafeAreaView>
+  )
+}
+
+function parseResultIntoSections(categoriesWithDapps: CategoryWithDapps[]): SectionData[] {
+  return categoriesWithDapps.map((category) => ({
+    data: category.dapps,
+    category: category,
+  }))
+}
+
+function DescriptionView({ message }: { message: string }) {
+  return (
+    <View style={styles.descriptionContainer}>
+      <Text style={{ ...fontStyles.h1, flex: 1 }}>{message}</Text>
+      <View style={styles.descriptionImage}>
+        <DappsExplorerLogo />
+      </View>
+    </View>
+  )
+}
+
+function CategoryHeader({ category }: { category: CategoryWithDapps }) {
+  return (
+    <View style={styles.categoryContainer}>
+      <View
+        style={{
+          ...styles.categoryTextContainer,
+          backgroundColor: category.backgroundColor,
+        }}
+      >
+        <Text style={{ ...styles.categoryText, color: category.fontColor }}>{category.name}</Text>
```

Comment:
i notice that there are a lot of inline styles in this component, do we want to keep them like this or move to a stylesheet? i think having styles in the stylesheet has some performance benefits (not recreating the style every rerender) and slightly easier to read but i see there are other places in the app where we use inline style so i'm not sure! just thought i'd ask, what do you think?

Reply:
We're trying to avoid inlining whenever possible. But I guess here it would look like this, which is really equivalent:
```suggestion
        <Text style={[styles.categoryText, { color: category.fontColor }]}>{category.name}</Text>
```